test/integration/targets/mysql_replication/tasks/mysql_replication_initial.yml

```diff
@@ -44,12 +46,13 @@
     master_password: "{{ replication_pass }}"
     master_log_file: "{{ master_status.File }}"
     master_log_pos: "{{ master_status.Position }}"
+    master_ssl_ca: ' '
```

Comment:
Requiring a string of exactly one space is a bit fragile. I would suggest using `None` or an empty string in the code as an indicator that it should be an empty string to make the interface a bit more flexible. Then the module interface would look like this:

```yaml
master_ssl_ca: 
master_ssl_ca: ~
master_ssl_ca: null
master_ssl_ca: ''
```

You would then probably want to change the default value to something else to give yourself an indication as to whether a value was passed into the playbook on not.

```python
master_ssl_ca=dict(type='str', default='__unset__')
```

Reply:
@samdoran good suggestion! will do

Reply:
Done, @samdoran , PTAL (i chose an underscore because there are different checks in the code - a part for None, others for True/False)